test/unit/specs/core/createEventManager.spec.js

```diff
@@ -0,0 +1,223 @@
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+// eslint-disable-next-line no-unused-vars
+import createEventManager from "../../../../src/core/createEventManager";
+// import createPayload from "../../../../src/core/network/createPayload";
+// import { defer } from "../../../../src/utils";
+// import flushPromiseChains from "../../helpers/flushPromiseChains";
+
+// const logger = {
+//   log() {},
+//   warn() {}
+// };
+const config = {
+  imsOrgId: "ABC123",
+  get() {}
+};
+
+describe("createEventManager", () => {
+  let event;
+  let payload;
+  let lifecycle;
+  let network;
+  let optIn;
+  // let onBeforeEventSpy;
+  // let onBeforeDataCollectionSpy;
+  // let sendRequestSpy;
+  let eventManager;
+  beforeEach(() => {
+    event = { mergeXdm() {}, isDocumentUnloading: () => true };
+    const createEvent = jasmine.createSpy().and.returnValue(event);
+    lifecycle = {
+      onBeforeEvent: jasmine.createSpy().and.returnValue(Promise.resolve()),
+      onBeforeDataCollection: jasmine
+        .createSpy()
+        .and.returnValue(Promise.resolve())
+    };
+    payload = {
+      addEvent: jasmine.createSpy(),
+      mergeMeta: jasmine.createSpy(),
+      expectsResponse: true
+    };
+    network = {
+      createPayload: () => payload,
+      sendRequest: jasmine.createSpy().and.returnValue(Promise.resolve({}))
+    };
+    optIn = {
+      whenOptedIn: () => Promise.resolve()
+    };
+    eventManager = createEventManager({
+      createEvent,
+      optIn,
+      lifecycle,
+      network,
+      config
+    });
+  });
+
+  describe("createEvent", () => {
+    it("creates an event object", () => {
+      expect(eventManager.createEvent()).toBe(event);
+    });
+  });
+
+  describe("sendEvent", () => {
+    it("calls onBeforeEvent", () => {
+      const options = {
+        isViewStart: true
+      };
+      return eventManager.sendEvent(event, options).then(() => {
+        expect(lifecycle.onBeforeEvent).toHaveBeenCalledWith({
+          event,
+          isViewStart: true
+        });
+      });
+    });
+
+    it("applies user provided data", () => {
+      const options = {
+        applyUserProvidedData: jasmine.createSpy()
+      };
+      return eventManager.sendEvent(event, options).then(() => {
+        expect(options.applyUserProvidedData).toHaveBeenCalledWith(event);
+      });
+    });
+
+    it("calls onBeforeDataCollection", () => {
+      return eventManager.sendEvent(event).then(() => {
+        expect(lifecycle.onBeforeDataCollection).toHaveBeenCalledWith({
+          payload
+        });
+      });
+    });
+
+    it("calls sendRequest", () => {
+      return eventManager.sendEvent(event).then(() => {
+        expect(network.sendRequest).toHaveBeenCalledWith(payload, true, true);
+      });
+    });
+
+    // it("resolves with response information", () => {
```

Comment:
Should we uncomment those lines? or remove them?

Reply:
I think you were commenting on the https://github.com/adobe/alloy/pull/253/files/61f641f7fec32013eca3253081596f7012cee294#diff-34fae68cfdead6324f76042dd9610713 commit directly instead of the latest. This was updated in a later commit.